src/main/java/edu/gatech/cs2340/trydent/Mouse.java

```diff
@@ -0,0 +1,144 @@
+package edu.gatech.cs2340.trydent;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javafx.scene.input.MouseButton;
+import javafx.scene.input.MouseEvent;
+
+/**
+ * A static class for ease of using mouse events.
+ *
+ */
+public class Mouse {
+    private static double x;
+    private static double y;
+
+    private static Set<MouseButton> down = new HashSet<>();
+    private static Set<MouseButton> downBuffer = new HashSet<>();
+    private static Set<MouseButton> pressed = new HashSet<>();
+    private static Set<MouseButton> pressedBuffer = new HashSet<>();
+    private static Set<MouseButton> released = new HashSet<>();
+    private static Set<MouseButton> releasedBuffer = new HashSet<>();
+
+    private static final Object POSITION_LOCK = new Object();
+    private static final Object REPLACE_LOCK = new Object();
+    private static final Object UPDATE_LOCK = new Object();
+
+    /**
+     * Gets the absolute position of the mouse with left as zero.
+     * @return the x position
+     */
+    public static double getMouseX() {
+        synchronized(POSITION_LOCK) {
```

Comment:
Having a lock for accessing a double variable seems really excessive. A double won't throw a ConcurrentModificationException if it's written to and read at the same time, so the lock just slows things down.